RadixWallet/Clients/ProfileStore/ProfileStore.swift

```diff
@@ -185,100 +78,142 @@ extension ProfileStore {
 		} catch {
 			let errorMessage = "Error, failed to delete profile or factor source, failure: \(String(describing: error))"
 			loggerGlobal.error(.init(stringLiteral: errorMessage))
 			assertionFailure(errorMessage)
 		}
-		let ephemeral = await Self.newEphemeralProfile()
-		changeState(to: .ephemeral(.init(profile: ephemeral, loadFailure: nil)))
+
+		let profile = try! Self._tryGenerateAndSaveNewProfile(deviceInfo: deviceInfo)
+		self.profileSubject.send(profile)
+	}
+
+	public func finishedOnboarding() async {
+		@Dependency(\.device) var device
+		if !profile.hasMainnetAccounts() {
+			let errorMsg = "Incorrect implementation should have accounts on mainnet after finishing onboarding."
+			loggerGlobal.error(.init(stringLiteral: errorMsg))
+			assertionFailure(errorMsg)
+		}
+		let model = await device.model
+		let name = await device.name
+		let deviceDescription = DeviceInfo.deviceDescription(
+			name: name,
+			model: model
+		)
+		deviceInfo.description = deviceDescription
+		let lastUsedOnDevice = deviceInfo
+		try? secureStorageClient.saveDeviceInfo(lastUsedOnDevice)
+		try? await updating {
+			$0.header.lastUsedOnDevice = lastUsedOnDevice
+			$0.header.creatingDevice.description = deviceDescription
+		}
+	}
+}
+
+extension DeviceInfo {
+	public static func deviceDescription(
+		name: String,
+		model: String
+	) -> String {
+		"\(model) (\(name))"
 	}
 }
 
+// MARK: - ProfileStore.Error
 extension ProfileStore {
-	// Changes the currently used ProfileSnapshot, usually to one from a backup or to one just created.
-	func changeProfileSnapshot(to profileSnapshot: ProfileSnapshot) async throws {
-		var profileSnapshot = profileSnapshot
-		profileSnapshot.changeCurrentToMainnetIfNeeded()
-		try await claimProfileSnapshot(&profileSnapshot)
-		updateHeader(&profileSnapshot)
-
-		// Save the updated snapshot.
-		// Do not check the ownership since the device did claim the profile ownership.
-		try await saveProfileSnapshot(profileSnapshot, checkOwnership: false)
-		// Update to new active profile id, so it is used from now on.
-		await userDefaultsClient.setActiveProfileID(profileSnapshot.header.id)
-
-		// Update the state with the imported snapshot
-		try changeState(to: .persisted(.init(snapshot: profileSnapshot)))
-	}
-
-	/// Claim the profile by updating **lastUsedOnDevice**
-	func claimProfileSnapshot(_ snapshot: inout ProfileSnapshot) async throws {
-		snapshot.header.lastUsedOnDevice = try await Self.createDeviceInfo()
-		do {
-			try await secureStorageClient.saveDeviceIdentifierIfNeeded(snapshot.header.lastUsedOnDevice.id)
-		} catch {
-			loggerGlobal.critical("Failed to save newly generated device identifier, error: \(error)")
+	public enum Error: String, Swift.Error, Equatable {
+		case profileIDMismatch
+		case profileUsedOnAnotherDevice
+	}
+}
+
+// MARK: "Private"
+extension ProfileStore {
+	func _lens<Property>(
```

Comment:
Drawback of Xcode only project (mono target) is that `internal` suddenly is visible to the whole app... which we dont want. And since Swift Core Team forwhatever reason still do not wanna give use `typeprivate` access modifier, we have to retort to `internal` access modifiers when we split methods between files.... :/ :/ :/ I use `_` prefix to mark that it is not something other types should call